src/main/java/com/carhouse/provider/impl/CarSaleProviderImpl.java

```diff
@@ -86,27 +71,22 @@ public void addCarSale(final CarSale carSale, final int[] carFeatures) {
     }
 
     /**
-     * this is a temporary object to facilitate the creation of CarSaleDto.
+     * Build url with query parameters.
+     * Validates incoming parameters and adds only valid
      *
-     * @param carSaleId car sale id
-     * @param carMake   car make
-     * @param carModel  car model
-     * @return car sale stub object
+     * @param baseUrl is url without query parameters
+     * @param requestParams request parameters
+     * @return url with query parameters
      */
-    private CarSaleDto createCarSaleDto(final Integer carSaleId, final String carMake, final String carModel) {
-        return new CarSaleDto()
-                .setCarSaleId(carSaleId)
-                .setPrice(new BigDecimal(30200))
-                .setDate(Date.valueOf("2019-03-02"))
-                .setCar(new CarDto()
-                        .setYear(Date.valueOf("2017-01-01"))
-                        .setMileage(140000)
-                        .setFuelType(new FuelType(1, "Bensin"))
-                        .setTransmission(new Transmission(0, "Manual"))
-                        .setCarModel(new CarModel()
-                                .setCarModel(carModel)
-                                .setCarMake(new CarMake(0, carMake))
-                        )
-                );
+    private String buildUrl(final String baseUrl, final Map<String, String> requestParams) {
+        HashMap<String, String> map = new HashMap<>(requestParams);
```

Comment:
`Map<String, String> map = new HashMap<>(requestParams);`

Reply:
I am using this because Map is not iterable